File: app/src/main/java/com/applications/whazzup/photomapp/di/modules/LocalModule.java
package com.applications.whazzup.photomapp.di.modules;


import com.applications.whazzup.photomapp.data.managers.PreferencesManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class LocalModule {
    @Provides
    @Singleton
    PreferencesManager providePreferencesManager(){
        return new PreferencesManager();
    }
}

File: app/src/main/java/com/applications/whazzup/photomapp/di/modules/LocalModule.kt
package com.applications.whazzup.photomapp.di.modules


import com.applications.whazzup.photomapp.data.managers.PreferencesManager
import com.applications.whazzup.photomapp.data.managers.RealmManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class LocalModule {
    @Provides
    @Singleton
    internal fun providePreferencesManager(): PreferencesManager {
        return PreferencesManager()
    }

    @Provides
    @Singleton
    internal fun provideRealmManager(): RealmManager {
        return RealmManager()
    }

}
